modin/pandas/dataframe.py

```diff
@@ -459,6 +459,8 @@ def groupby(
             if (
                 self._query_compiler.has_multiindex(axis=axis)
                 and by in self.axes[axis].names
+                or hasattr(self.axes[axis], "name")
```

Comment:
@devin-petersohn We try to avoid index access here to enable lazy groupby (and `has_multiindex` was added for that). With this patch, we lose lazy groupby again.

Reply:
Since this was merged, can we open and discuss in a new issue? Discussion can get lost otherwise. I agree this is a problem.